lib/hawkular/inventory/inventory_api_v4.rb

```diff
@@ -37,35 +36,47 @@ def self.create(hash)
     # Get single resource by id
     # @return Resource the resource
     def resource(id)
-      hash = http_get("/resources/#{URI.escape(id)}")
+      hash = http_get(url('/resources/%s', id))
       Resource.new(hash)
     end
 
     # Get resource by id with its complete subtree
     # @return Resource the resource
     def resource_tree(id)
-      hash = http_get("/resources/#{URI.escape(id)}/tree")
+      hash = http_get(url('/resources/%s/tree', id))
       Resource.new(hash)
     end
 
     # Get childrens of a resource
     # @return Children of a resource
     def children_resources(parent_id)
-      http_get("/resources/#{parent_id}/children")['results'].map { |r| Resource.new(r) }
+      http_get(url('/resources/%s/children', parent_id))['results'].map { |r| Resource.new(r) }
     end
 
     # List root resources
     # @return [Array<Resource>] List of resources
     def root_resources
       # FIXME: pagination => lazy-loaded list with ruby?
-      http_get('/resources?root=true')['results'].map { |r| Resource.new(r) }
+      resources root: "true"
+    end
+
+    # List resources
+    # @param [Hash] filter options to filter the resource list
+    # @option filter :root If truthy, only get root resources
+    # @option filter :feedId Filter by feed id
+    # @option filter :typeId Filter by type id
+    def resources(filter = {})
+      # FIXME: pagination => lazy-loaded list with ruby?
+      filter[:root] = !! filter[:root] if filter.key? :root
+      filter_query = '?' + filter.keys.join('=%s&') + '=%s' unless filter.empty?
+      http_get(url("/resources#{filter_query}", *filter.values))['results'].map { |r| Resource.new(r) }
```

Comment:
@jotak can you review this part?

Reply:
I've no strong opinion about opening the API to queries with feedId, however note that I didn't put a "get_by_feedid" function in a first place, because I think MiQ should be able to perform all of its queries without having to deal with the feedId. (Except for operations, that's why it's still in the model)

According to Lucas, the feedId leaking in MiQ code was more of a technical debt due to early versions. If we can run queries without it, we would mechanically reduce the number of calls.

Reply:
(but the ruby client is not just for MiQ so I think it's fine)